packages/transaction-controller/src/utils/validation.ts

```diff
@@ -281,6 +286,14 @@ function validateGasFeeParams(txParams: TransactionParams) {
     );
     ensureFieldIsValidHex(txParams, 'maxPriorityFeePerGas');
   }
+
+  if (txParams.gasLimit) {
```

Comment:
Probably a future ticket, but we should make sure we also normalize these together to a single property so we can rely on a single value in the rest of the logic.

Reply:
It would be a huge relief, these optional picks are all over the client code

Reply:
We could even make that a caller problem and only accept one in `TransactionParams`.

Reply:
This is a great idea and I agree with normalizing these. Shouldn't we leave `gasLimit` in the transaction params since the incoming payload from the dApps still might include it?

Reply:
I meant "caller" as in the client rather than the dApp. So the client middleware could merge the properties and give us just one.